module-dir-structure.md

```diff
@@ -1,37 +1,236 @@
+
+Although the slides are written in Markdown, they are rendered using [Reveal.js](https://revealjs.com/), and for this reason, they follow a special format. This is discussed below.
+
+### Slide types
+
+There are four different slide types:
+
+- Title slide, `data-state="title"`
+- Default slide, `data-state="standard"`
+- "About us", `data-state="about"`
+- "Let's stay in touch", `data-state="keepintouch"`
+
+A slide is fenced by three dashes, and (optionally) an HTML comment that indicates the slide type:
+
+```markdown
+
+---
+
+<!-- .slide: data-state="standard" -->
+
+```
+
+Always keep an empty line before and after the slide fence.
+
+The dashes indicate the slide borders; the are therefore only necessary between the slides, and not at the beginning or end of a presentation.
+
+### Slide content
+
+Slide content can be written in Markdown.
+
+To keep slides clean, use single images per slide.
+Styling in Reveal is not trivial, and it is best to keep it simple.
+
+Images can be embedded using either markdown syntax:
+
+```markdown
+![Mapping the Via Appia](media/viaappia.png)
+```
+
+Or, for more customisation, using HTML:
+
+```html
+<center>
+<img src="media/researchcycle.png" width="60%">
+</center>
+```
+
+### Slide notes
+
+Notes should be added at the bottom of the slide, as follows:
+
+```markdown
+
+Note:
+Here is the text of a note.
+
+---
+```
+
+where the `---` indicates the fence to the following slide.
+
+### Final slide
+
+The final slide should provide the contact information for the eScience Center.
+This is not hardcoded into the slides, so it should be provided explicitly.
+
+The code for the final slide is as follows:
+
+```markdown
+
+---
+
+<!-- .slide: data-state="keepintouch" -->
+
+
+www.esciencecenter.nl
 
-## The slides pmd file
+info@esciencecenter.nl
 
-## Exercises and online resources files
+020 - 460 47 70
+
+```
+
+## Text, exercises and online resources files
+These chapters are defined by simply Markdown `*.md` files with an initial YAML heading.
+
+An example `text.md` file is shown below:
+```markdown
+---
+title: Software Testing
+type: reading
+order: 2
+---
+
+## Software Testing (5 minutes)
+
+Software testing is the process of evaluating and verifying that software meet specified requirements...
+```
+where it uses the `reading` chapter type as described [above](#the-infomd-file).
+In turn, an `exercise.md` chapter can assume the following format:
+
+```markdown
+---
+title: Exercise 1
+type: exercise
+order: 3
+---
+
+# Exercise 1
+
+....
+```
+
+while an online resources file is given by:
+
+```markdown
+---
+title: Reading material
+type: reading
+order: 4
+---
+
+# Reading material
+
+## Testing
+
+Follow the links below to read more about software testing.
+....
+```
 
 ## The `media` folder
 
+The `media` folder gathers any media used in your module, including presentation images, videos, and module's thumbnail image.
```

Comment:
Also good to mention how you then point to these media files from any of the md files.

Reply:
Done.